### ISSUE TITLE
Timeout issues with poloniex

### ISSUE THREAD

**Issue by offtopic** [MarcelBeining/EazeBot#6]
First of all, the bot is really cool and helps a lot to track and handle orders!

**Describtion of the issue**
Telegram pushes in irregular intervals an error message, which makes it difficult to have the focus on the real trades

```
eazebot, [13.12.18 00:35]
error: Waiting for tradeSet update to finish timed out after 1 min, resetting updating variable
...
eazebot, [13.12.18 00:45]
eazebot, [13.12.18 00:48]
eazebot, [13.12.18 01:16]
eazebot, [13.12.18 01:23]
eazebot, [13.12.18 02:13]
...
```

**telegramEatzeBot.log**
```
2018-12-13 00:33:30,461 [job_queue   ] [INFO ]  Updating trade sets...
2018-12-13 00:33:49,867 [job_queue   ] [ERROR]  poloniex POST https://poloniex.com/tradingApi HTTPSConnectionPool(host='poloniex.com', port=443): Read timed out. (read timeout=10)
2018-12-13 00:33:49,867 [job_queue   ] [INFO ]  Finished updating trade sets...
2018-12-13 00:34:00,461 [job_queue   ] [INFO ]  Updating trade sets...
2018-12-13 00:35:04,709 [job_queue   ] [ERROR]  User 479767609: Waiting for tradeSet update to finish timed out after 1 min, resetting updating variable
2018-12-13 00:35:04,755 [job_queue   ] [WARNI]  Some connection (?) error occured when trying to send a telegram message. Retrying..
2018-12-13 00:35:12,149 [job_queue   ] [INFO ]  Finished updating trade sets...
```
Is the read timeout of "10" the issue?

It only seems related to POLONIEX, I have also used the bot without POLO -> no issues

**System**
 - OS: Windows 10
 - EazeBot (v1.7.9) *
 - ccxt 1.18,5

**I only have changed the updateInterval to 0.5 in the botConfig.json**
^^ Is this a problem?

Regards Off

**Comment by offtopic** [comment]
Thanks for your fast and detailed reply!

Sould i try to update the read timeout of 10 seconds?

> However I will try to find ways (e.g. catching even more connection errors) to minimize these messages to a minimum and let you know about it.
That would be nice!

Best regards
Off

**Comment by offtopic** [comment]
I'm working on a windows client to handle and visualize running trades, are you interested to integrate your bot into my GUI?

![image](https://user-images.githubusercontent.com/3337241/49963045-37c83b00-ff17-11e8-8d70-5e8951e335a2.png)

developed with c#/wpf

**Comment by MarcelBeining** [comment]
> Thanks for your fast and detailed reply!
> 
> Sould i try to update the read timeout of 10 seconds?
> 
> > However I will try to find ways (e.g. catching even more connection errors) to minimize these messages to a minimum and let you know about it.
> > That would be nice!
> 
> Best regards
> Off

It could be that you can change this when initializing a the ccxt exchange object in tradeHandler, however longer read timeouts also mean your bot takes longer to respond if some connection makes problems. I'd rather suggest you to increase the updateInterval in botConfig.json and wait for updates from me :-)

Concerning your request: Nice GUI! If you want you can write to me @ marcel.beining@gmail.com what your idea is concerning implementation.

**Comment by offtopic** [comment]
Sorry, forgot the log

`2018-12-21 15:32:52,421 [job_queue   ] [ERROR]  User 479767609: Waiting for tradeSet update to finish timed out after 1 min, resetting updating variable
C:\Users\thoma\AppData\Local\Programs\Python\Python36-32\lib\site-packages\numpy\core\fromnumeric.py:2920: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\thoma\AppData\Local\Programs\Python\Python36-32\lib\site-packages\numpy\core\_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
2018-12-21 15:33:55,178 [job_queue   ] [ERROR]  User 479767609: Waiting for tradeSet update to finish timed out after 1 min, resetting updating variable`

**Comment by MarcelBeining** [comment]
Hi offtopic,
yes I just uploaded a new version [1.7.13](https://github.com/MarcelBeining/EazeBot/releases/tag/EazeBot_1.7.13), which should at least solve one problem causing this..
Actually the problem was that the updating variable was autosaved, too, meaning when you stopped the bot during an update it could happen that the updating was set to True and the waiting queue was non-empty at the moment of saving which then caused problems when starting the bot.
Actually I hoped to reduce the updating message problem when switching from exchange updating to tradeSet updating variable but it had multiplied :-D Really sorry about that, I will investigate it further...

**Comment by MarcelBeining** [comment]
Seems the issue is finally solved with [1.7.14](https://github.com/MarcelBeining/EazeBot/releases/tag/EazeBot_1.7.14)!  =)

The issue was one line having a wrong indent  **argh**

**Comment by offtopic** [comment]
Thank you very much for your fast response & fix!
Great work!
Regarding my trading client -> I have to sort some things out and I will come back to you.
